runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcGetDataStreamRequests.java

```diff
@@ -95,11 +108,8 @@ static class QueuedBatch {
     private final List<QueuedRequest> requests = new ArrayList<>();
```

Comment:
Comments on which methods are thread safe and which ones need external synchronization would help.

Reply:
done